backend/src/controllers/DailyWorkoutsController.js

```diff
@@ -24,7 +28,89 @@ const getDailyWorkoutsbyDate = async (req, res) => {
     }
 }
 
+
+  const insertWorkout = async (req, res) => {
+    const userId = req.user.data.user_id;
+    const { dailyworkout_id, workout_id, sets, reps } = req.body;
+    const currentDate = new Date(); // Get the current date 
+    console.log("hit");
+    try {
+        const workoutExists = await db.oneOrNone(
+            "SELECT * FROM dailyworkouts WHERE dailyworkout_id = $1",
+            [dailyworkout_id]
+        );
+
+        if (!workoutExists) {
+            const insertedWorkout = await db.one(
+                "INSERT INTO dailyworkouts(dailyworkout_id, user_id, workout_id, sets, reps, date) VALUES($1, $2, $3, $4, $5, $6) RETURNING *",
+                [dailyworkout_id, userId, workout_id, sets, reps, currentDate]
+            );
+            return res.status(200).json({ success: true, workout: insertedWorkout, message: "Workout inserted successfully" });
+        } else {
+            return res.status(400).json({ success: false, message: "Workout already exists" });
+        }
+    } catch (error) {
+        return res.status(500).json({ success: false, message: "Error inserting workout into the database: " + error.message });
+    }
+};
+
+
+const updateDailyWorkout = async (req, res) => {
+  const { dailyworkout_id, user_id, workout_id, sets, reps } = req.body;
```

Comment:
I believe we should only allow users to update their `reps` and `sets` for a given `dailyworkout_id` therefore, we don't need to get the `workout_id`